integrators/integrators_body.hpp

```diff
@@ -12,10 +12,86 @@
 #include "integrators/symmetric_linear_multistep_integrator.hpp"
 #include "integrators/symplectic_runge_kutta_nyström_integrator.hpp"
 
+#define PRINCIPIA_CASE_SLMS(kind, method)                      \
+  case serialization::FixedStepSizeIntegrator::kind:           \
+    return SymmetricLinearMultistepIntegrator<methods::method, \
+                                              typename ODE::Position>()
+
+// We do not deserialize an SPRK per se, but only when it is converted to an
+// SRKN.  The reason is that an SPRK is for a different kind of equation than
+// an SRKN, so the two would return different types.  If we ever need to do
+// this we will need to specialize ReadFromMessage somehow.
+#define PRINCIPIA_CASE_SPRK(kind, method)                                 \
+  case serialization::FixedStepSizeIntegrator::kind:                      \
+    do {                                                                  \
+      CHECK(message.has_composition_method());                            \
+      return SprkAsSrknDeserializer<                                      \
+          ODE,                                                            \
+          methods::method,                                                \
+          methods::method::first_same_as_last>::ReadFromMessage(message); \
+    } while (true)
```

Comment:
Do we really want `while (true)`?
It doesn't matter because of the return, but it seems odd.

Reply:
Silly me.